app/Models/Generic.php

```diff
@@ -39,20 +43,101 @@ static function fetchPosts()
 		return $result;
 	}
 
-	// return number of posts by counting the number of rows
+	static function fetchSearchResults($search)
+	{
+		$db = Connection::connect();
+		$sql = "SELECT * FROM cms.posts WHERE post_tags LIKE '%{$search}%'";
+		$query = mysqli_query($db, $sql);
+
+		if ( ! $query) {
+			echo mysqli_error($query);
+		}
+
+		$result = [];
+		while ($row = mysqli_fetch_assoc($query)) {
+			$result[] = $row;
+		}
+
+		return $result;
+	}
+
+	static function fetchPost($id)
+	{
+		$db = Connection::connect();
+		$sql = "SELECT * FROM cms.posts WHERE cms.posts.id = {$id}";
+		$query = mysqli_query($db, $sql);
+
+		if ( ! $query) {
+			echo mysqli_error($query);
+		}
+
+		$row = mysqli_fetch_assoc($query);
+
+		return $row;
+	}
+
 	static function fetchPostCount()
 	{
 		$db = Connection::connect();
-		$sql = "SELECT * FROM cms.posts";
-		$find_count = mysqli_query($db, $sql);
-		$count = mysqli_num_rows($find_count);
-		$count = ceil($count / 5); // make int, not float
-		return $count;
+		$sql = "SELECT * FROM csm.posts";
```

Comment:
Typo in the database name. You could consider cleaning up your queries even more by not referencing the database name at all. Since your database is already connected to a name this is redundant and cleans up the visual ease of reading this query.

`SELECT * from posts...`

Reply:
That said, you are binding your code to a named storage which seems like a weird coupling. With you moving away from referencing your db names in the queries, you can name your database to anything you want. 

Sometimes beneficial when it comes to larger databases and you would want some backuos to run against etc.